### PR TITLE
Handling BooleanOptionSetToBoolean coercion

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx-Dataverse/issues/61

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs [hunk 1]
```diff
@@ -449,6 +460,12 @@ public static FormulaValue OptionSetValueToString(IRContext irContext, OptionSet
             var displayName = optionSet.DisplayName;
             return new StringValue(irContext, displayName);
         }
+
+        public static FormulaValue BooleanOptionSetToBoolean(IRContext irContext, OptionSetValue[] args)
+        {
+            var arg0 = args[0];
+            return arg0.Option == "1" ? FormulaValue.New(true) : FormulaValue.New(false);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs:467]: I wonder if we should throw / error if it's not "1" or "0".

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by ian-legler** [comment]
Should there be some unit tests to cover these changes?

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs (hunk 1 above)
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs:467]: Good question. From DV we know we can expect "1" or "0". I'm wondering if we should check for "true"/"false".

**Review by ian-legler — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**